### PR TITLE
Fix typos in build.sh

### PR DESCRIPTION
**Pull request by mellinoe**
Fixes ~~a couple~~ one problem:

* Parameter wasn't quoted, so if nothing is passed to the script, there's a syntax error.
* ~~The build still runs even if you pass `-?`~~

@maririos @Priya91

### CODE REVIEW

**Comment by maririos** [comment]
Thanks @mellinoe ! LGTM

File: build.sh [hunk 1]
```diff
@@ -17,8 +17,9 @@ usage()
     echo
 }
 
-if [ $1 == "-?" ]; then
+if [ "$1" == "-?" ]; then
     usage
+    exit 0
```
Comment (maririos) [inline build.sh]: hmm now that I think more about it, if we want to have the same behavior as build.cmd we need to keep running the script and pass `-?` to `build-native` and `build-managed`.

Comment (mellinoe) [inline build.sh]: Huh, I guess that is an option. By the way, build-managed.sh does not do anything with the `-?` parameter right now, even though there is a function to print out the usage. Any objection to just merging this as-is, and perhaps fixing that separately?

Comment (maririos) [inline build.sh]: I'll prefer if you delete the exit code here. Just for consistency.
I'll work in the `build-managed.sh` usage separately

Comment (mellinoe) [inline build.sh]: I deleted the early exit there.

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Comment by mellinoe** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please

**Pull request merged**